backend/resources/reviews.py

```diff
@@ -22,15 +21,21 @@ def parse_review():
     return parser.parse_args()
 
 
-def check_user_and_book(user_model, isbn):
+def check_user_and_book(user_model, isbn, ignore_if_admin=False):
     if not user_model:
         abort(404, message={"message": f"User Not Found"})
 
     if not BooksModel.find_by_isbn(isbn):
         abort(404, message={"message": f"Book with ['isbn': {isbn}] Not Found"})
 
-    if g.user != user_model:
-        abort(401, message={"message": "Invalid user to remove, can only be yourself"})
+    if ignore_if_admin:
+        # First statement checks if the role is user and that it doesn't try to modify an other user
+        # Second statement checks if the user who wants to modify it's not and Admin
+        if (g.user.role is not Roles.User or g.user != user_model) and g.user.role is not Roles.Admin:
```

Comment:
No entenc prou bé aquesta condició. Me la pots explicar?

Reply:
- `(g.user.role is not Roles.User or g.user != user_model)` comprova que si el usuari logejat en cas de ser rol user, no intenti modificar el rol de un altre.  Esta abreviat tal vegada ho entes millor aixi `not (g.user.role is Roles.User and g.user == user_model)`.
- `g.user.role is not Roles.Admin` si el usuari logeajat es rol admin, no comprovara que intenti modificar un altre usuari.

Reply:
Sí ara molt millor! Gràcies!